cppcache/include/geode/DataInput.hpp

```diff
@@ -341,6 +339,11 @@ class CPPCACHE_EXPORT DataInput {
     str[length] = '\0';
   }
 
+  template <class CharT, class... Tail>
+  inline void readAscii(std::basic_string<CharT, Tail...>& value) {
```

Comment:
I use that to enforce that it must be a `basic_string` type but it isn't necessary. Either way the compiler will fail, either that it can't match a template or it can't find a string method on some random provided type.

Reply:
I think in other cases I needed to extract the character type and this makes that easy.